app/livechat/server/lib/Helper.js

```diff
@@ -393,7 +393,7 @@ export const forwardRoomToDepartment = async (room, guest, transferData) => {
 
 	Livechat.saveTransferHistory(room, transferData);
 	if (oldServedBy) {
-		removeAgentFromSubscription(rid, oldServedBy);
+		RoutingManager.removeAllRoomSubscriptions(room);
```

Comment:
Here you're removing all subscriptions, including the subscription of the agent who was assigned to the chat through the `RoutingManager.takeInquiry` method.

![Screen Shot 2021-06-16 at 6 23 23 PM](https://user-images.githubusercontent.com/59577424/122295761-fb9f3800-cecf-11eb-8265-877f7a250b2b.png)

Also, the room has been closed right after the transfer process took place, so I don't know if this is an issue introduced now.

Reply:
Handled this by adding a new param (`ignoreUser`) to `RoutingManager.removeAllRoomSubscriptions`. Then while forwarding we pass the newly assigned user to this method as `ignoreUser` and so the new subscription will not be deleted.

Regarding the issue of - closing of chat after transfer - I wasn't able to reproduce it. In my testing after transferring to an agent or department - the chat is not closed automatically. Could you check if your `Visitor Abandonment` settings isn't closing the chat?